### PR TITLE
fix skipped Test: TraceProviderSdkTest.TracerProviderSdkInvokesSamplingWithCorrectParameters

### LINKED ISSUE #2808: Skipped Test: TraceProviderSdkTest.TracerProviderSdkInvokesSamplingWithCorrectParameters
I'm looking into a unit test that is marked `skip`. 
I'm reviewing if the expected behavior is valid.

## Skip = "Get around GitHub failure"

https://github.com/open-telemetry/opentelemetry-dotnet/blob/aad309b9445e19c437f05df150af92227e1b9eb3/test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs#L37-L38

I don't know what this comment is referring to.
I see it was introduced in this PR: https://github.com/open-telemetry/opentelemetry-dotnet/pull/2174/files#r674458207

## Which Assert is failing?

The very last assert is failing. 
The comment suggests that because a non W3C parent ID has been provided (`"InvalidW3CIdParent"`) no activity would be created.
However, this expected behavior is not documented for the [system.diagnostics.activitysource.startactivity](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.activitysource.startactivity?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(System.Diagnostics.ActivitySource.StartActivity)%3Bk(DevLang-csharp)%26rd%3Dtrue&view=net-6.0#system-diagnostics-activitysource-startactivity(system-string-system-diagnostics-activitykind-system-string-system-collections-generic-ienumerable((system-collections-generic-keyvaluepair((system-string-system-object))))-system-collections-generic-ienumerable((system-diagnostics-activitylink))-system-datetimeoffset)) method.

I verified that this behavior is consistent for all 4 tested frameworks (net461, net5.0, net6.0, netcoreapp3.1).

https://github.com/open-telemetry/opentelemetry-dotnet/blob/aad309b9445e19c437f05df150af92227e1b9eb3/test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs#L101-L108

Debugging shows that a non-null `Activity` is created, but the `Parent` and `ParentId` are `null`:
image

## Working example of StartActivity

For comparison, the previous `using` block calls `StartActivity` and provides a W3C parentid (`customContextAsString = "00-263e96dabbdbe6822b9b270d9cb15176-9160fdf894bed8cd-00"`).
https://github.com/open-telemetry/opentelemetry-dotnet/blob/aad309b9445e19c437f05df150af92227e1b9eb3/test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs#L93-L99

image

## Suggested Fix

I verified that the behavior is consistent for all 4 tested frameworks (net461, net5.0, net6.0, netcoreapp3.1).
We could enable this test if we change the asserts:

```
            using (var fromInvalidW3CIdParent =
                activitySource.StartActivity(name: "customContext", kind: ActivityKind.Client, parentId: "InvalidW3CIdParent"))
            {
                Assert.NotNull(fromInvalidW3CIdParent);
                Assert.Equal("customContext", fromInvalidW3CIdParent.DisplayName);
                Assert.Equal(ActivityKind.Client, fromInvalidW3CIdParent.Kind);
                // OpenTelemetry ActivityContext does not support non W3C Ids. 
                // Starting activity with non W3C Ids will result in an activity with no ParentId being created.
                Assert.Null(fromInvalidW3CIdParent.ParentId);
            }
```

### PR DESCRIPTION
**Pull request by TimothyMothra**
Fixes #2808 .

## Changes
- enable skipped test
- former code expected a null `Activity`. In contrast, an `Activity` is created with a null `ParentId`. 

Please provide a brief description of the changes here.

former code expected a null `Activity`. In contrast, an `Activity` is created with a null `ParentId`. 
Full explanation with debugging screenshots is in the linked issue.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs [hunk 1]
```diff
@@ -101,10 +101,13 @@ public void TracerProviderSdkInvokesSamplingWithCorrectParameters()
...
-                // non W3C Ids. Starting activity with non W3C Ids
-                // will result in no activity being created.
-                Assert.Null(fromInvalidW3CIdParent);
+                Assert.NotNull(fromInvalidW3CIdParent);
+                Assert.Equal("customContext", fromInvalidW3CIdParent.DisplayName);
+                Assert.Equal(ActivityKind.Client, fromInvalidW3CIdParent.Kind);
+
+                // OpenTelemetry ActivityContext does not support non W3C Ids.
+                // Starting activity with non W3C Ids will result in an Activity being created with no ParentId.
+                Assert.Null(fromInvalidW3CIdParent.ParentId);
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs:110]: this is not doing what the test is claiming to be doing (Sampler receives correct params). If this seems irrelevant, consider removing this whole section.

Comment (TimothyMothra) [reply test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs:110]: Sure, I was confused about what to Assert here.
The former code assumed the Activity would be null, so my asserts were focused on the Activity instance.

I debugged this again and verified that the Sampler was invoked and records the LatestSamplingParameters.
I'm updating this part.

**Review by cijothomas — APPROVED** [review]
